cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casInstitutionLoginView.jsp

```diff
@@ -34,29 +37,49 @@
     </section>
     <section class="row">
         <div class="select">
-            <form:select class="select" id="institution-form-select" name="select-institution" path="institutions" items="${institutions}" />
+            <form:select class="select" id="institution-form-select" name="select-institution" path="institutions"
+                         items="${institutions}"/>
         </div>
     </section>
     <br>
     <section class="row">
         <div class="btn-submit">
-            <input type="button" name="submit" value="<spring:message code="screen.welcome.button.login" />" onclick="institutionLogin()">
+            <input type="button" name="submit" value="<spring:message code="screen.welcome.button.login" />"
+                   onclick="institutionLogin()">
         </div>
     </section>
-    <br>
+
+    <%-- OSF Username and Password Login --%>
+    <br/>
+    <hr/>
+    <br/>
+    <spring:eval var="osfLoginUrl" expression="@casProperties.getProperty('cas.osf.login.url')"/>
+    <c:set var="serviceParam"
+           value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+    <section class="row">
```

Comment:
Added a button for "Non-institution Login" and removed the link from the `bottom.jsp`.